src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.trust.TrustDataDto;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.DateValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationMessages;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+@Component
+public class TrustDetailsValidator {
+
+    public Errors validate(List<TrustDataDto> trustDataDtoList, Errors errors, String loggingContext) {
+
+        validateDuplicateId(trustDataDtoList, errors, loggingContext);
+
+        for(TrustDataDto trustDataDto : trustDataDtoList) {
+            validateName(trustDataDto.getTrustName(), errors, loggingContext);
+            validateCreationDate(trustDataDto.getCreationDate(), errors, loggingContext);
+            validateUnableToObtainAllTrustInfo(trustDataDto.getUnableToObtainAllTrustInfo(), errors, loggingContext);
+        }
+
+        return errors;
+    }
+
+
+    private boolean validateDuplicateId(List<TrustDataDto> trustDataDtoList, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(OverseasEntitySubmissionDto.TRUST_DATA,
+                TrustDataDto.TRUST_ID_FIELD);
+        Set<String> findDuplicates = new HashSet<>();
+
+        for (TrustDataDto trustDataDto : trustDataDtoList) {
+            String trustId = trustDataDto.getTrustId();
+
+            if (!StringValidators.isNotBlank(trustId, qualifiedFieldName, errors, loggingContext)) {
+                return false;
+            }
+
+            if (!findDuplicates.add(trustDataDto.getTrustId().trim())) {
```

Comment:
the trim() could throw a null pointer if getTrustId() returns null. I would also add a null check on trustDataDto.getTrustId()

Reply:
thanks @markpit I feel a null check will be unnecessary at this line. Because we are already checking for null in the above lines of code. I am sharing the block of code here

`            if (!StringValidators.isNotBlank(trustId, qualifiedFieldName, errors, loggingContext)) {
                return false;
            }

            if (!findDuplicates.add(trustDataDto.getTrustId().trim())) `